client/app/src/main/java/com/microsoft/research/karya/utils/Result.kt

```diff
@@ -1,7 +1,28 @@
 package com.microsoft.research.karya.utils
 
+import com.microsoft.research.karya.R
+import com.microsoft.research.karya.data.exceptions.AccessCodeAlreadyUsedException
+import com.microsoft.research.karya.data.exceptions.IncorrectAccessCodeException
+import com.microsoft.research.karya.data.exceptions.IncorrectOtpException
+import com.microsoft.research.karya.data.exceptions.UnknownException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+
 sealed class Result {
   class Success<T>(val value: T) : Result()
-  class Error(val exception: Throwable) : Result()
+  class Error(val exception: Throwable) : Result() {
+    val errorMessageId =
+        when (exception) {
+          is UnknownHostException -> R.string.s_no_internet_or_server_down
+          is SocketTimeoutException -> R.string.s_no_internet_or_server_down
+          // Socket Timeout may also happen due to bad internet connection, maybe indicate in the
+          // string?
+          is AccessCodeAlreadyUsedException -> R.string.s_phone_number_already_used
+          is IncorrectAccessCodeException -> R.string.s_invalid_creation_code
+          is IncorrectOtpException -> R.string.s_invalid_otp
+          is UnknownException -> R.string.s_unknown_error
+          else -> R.string.s_unknown_error
+        }
+  }
```

Comment:
Please don't do this, this makes `Result` dependent on android resources and hence difficult to unit-test.

Reply:
This should probably be handled inside the fragment

Reply:
You will end up doing this in every Fragment. What you say we go ahead with extensions?

Reply:
I think it's okay to handle it in every fragment because
- Not every fragment will deal with all of these errors, we can divide exceptions into a sealed class based on what exception can occur for that particular feature
- Displaying an error should be a feature of the view layer of the application and each view should handle it's own error/success state.

Reply:
You can write an extension for this but that should still be only used in the view layer and not in the viewmodel / repository.